graphics/macgui/macmenu.cpp

```diff
@@ -1229,8 +1253,11 @@ bool MacMenu::mouseClick(int x, int y) {
 		setActive(true);
 
 		if (_wm->_mode & kWMModalMenuMode) {
-			draw(_wm->_screen);
-			eventLoop();
+			if (!forceClick)
+				draw(_wm->_screen);
```

Comment:
Ugly. call ::draw as virtual function, have it overloaded in MacPopupMenu, and then you do not need this flag

Reply:
right. I will do this